docs/core/integration-quality-scale/rules/iqs012.md

```diff
@@ -0,0 +1,69 @@
+---
+title: "IQS012 - Entities use has_entity_name = True"
+---
+
+IQS012 is as follows:
+
+Entities use has_entity_name = True
+
+## Reasoning
+
+`has_entity_name` is an entity attribute, which enables a new naming method.
+It is introduced to split the name of the entity from the name of the device or service.
+With `has_entity_name` set to `True`, the user will have a better experience on the device page, since not every entity is prefixed with the device name.
```

Comment:
This isn't true, since that behavior doesn't need this attribute. The frontend was doing that before we introduced this attribute.

It's more in general that we'll be able to show a better name of the entity to the user depending on the context where the name is shown.

Reply:
Yea I never actually knew why it was introduced and this was my assumption. Apart from enabling translations and future frontend changes (that we only now slowly start to see PRs pop up for on the frontend repo), what is a good reasoning why we want to use this?

Reply:
The reason is my last paragraph.